### DISCUSSION TITLE
Is it possible to route all traffic through shadowsocks?

### DISCUSSION THREAD

**Discussion by shadowsocksquestions** [shadowsocks/shadowsocks-rust#482] [Q&A] [answered]
Hi

So I am developing a php application and need to access some blocked APIs/scrape some pages from the php local server, on my pc. In the browser, I can use shadowsocks to access these pages/endpoints, however, php seems to be just using my standard connection, rather than the shadowsocks connection.

Here is an example of my config.json

```
{
    "server": "usmvip8.helike.me",
    "server_port": 4470,
    "password": "thepassword",
    "method": "aes-256-gcm",
    "local_address": "127.0.0.2",
    "local_port": 54321
}
```

So my questions are:

1) Is this possible? (I assume yes?)

2) What information should I use when trying to set up the proxy? For example I saw this code, which should supposedly route standard php functions through a proxy:

```php
stream_context_set_default([
'http'=>[
    'proxy'=>'proxy-host:proxy-port', 
    'header'=>'Proxy-Authorization: Basic '.base64_encode('your-username:your-password')
]
]);
```

For example, in the Ubuntu settings, I use the local address & port and that works...but here, should I put the server address:port or the local ones? I tried both, but neither worked (I didn't expect it would work that easily anyway...). Anyway, I don't want to write too much. I have also looked up PHP/Apache/Ubuntu proxy settings to try and get things working, but I guess I don't know if Shadowsocks works like a regular proxy...so don't know whether these approaches would work anyway

If someone has any advice for how I could get this working, I'd really appreciate it. 

Thanks

**Comment by zonyitoo** [comment, marked as answer]
With your configuration, `sslocal` serves SOCKS5 on `127.0.0.2:54321`. You could just set it as SOCKS5 proxy in your PHP environment.

**Reply by shadowsocksquestions** [reply]
Thanks. I will try this again. At least it is the local address:port that I should be using then, rather than the server ones in the config.json file. That makes it easier, since at least it's only 1...
